vcsim: unable to add host to DVS

1) Launch vcsim with default parameters:
```
[jonas@localhost bin]$ ./vcsim
export GOVC_URL=https://user:pass@127.0.0.1:8989/sdk GOVC_SIM_PID=4792
```
2) Create a new DVS:
```
[jonas@localhost bin]$ ./govc dvs.create dvSwitch
[17-11-17 09:30:59] adding dvSwitch to folder /DC0/network... OK
```
3) Try adding default host to DVS:
```
[jonas@localhost bin]$ ./govc dvs.add -dvs dvSwitch DC0_H0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0x9b702c]

goroutine 1 [running]:
github.com/vmware/govmomi/vim25/mo.anyTypeToValue(0x0, 0x0, 0xc4201ee960, 0x6, 0xc420285ac0)
	/home/jonas/.go/src/github.com/vmware/govmomi/vim25/mo/type_info.go:219 +0x9c
github.com/vmware/govmomi/vim25/mo.(*typeInfo).LoadFromObjectContent(0xc420263b00, 0xc4201e0e00, 0x18, 0xc4201e0ea0, 0x1b, 0xc4201f6780, 0x1, 0x4, 0x0, 0x0, ...)
	/home/jonas/.go/src/github.com/vmware/govmomi/vim25/mo/type_info.go:243 +0x241
github.com/vmware/govmomi/vim25/mo.ObjectContentToType(0xc4201e0e00, 0x18, 0xc4201e0ea0, 0x1b, 0xc4201f6780, 0x1, 0x4, 0x0, 0x0, 0x0, ...)
	/home/jonas/.go/src/github.com/vmware/govmomi/vim25/mo/retrieve.go:60 +0x15b
github.com/vmware/govmomi/vim25/mo.LoadRetrievePropertiesResponse(0xc4201e0d80, 0xe446c0, 0xc420291b80, 0x0, 0x0)
	/home/jonas/.go/src/github.com/vmware/govmomi/vim25/mo/retrieve.go:113 +0x6bb
github.com/vmware/govmomi/property.(*Collector).Retrieve(0xc4201cf908, 0x17dcb80, 0xc420014298, 0xc4202c2900, 0x1, 0x1, 0xc4202c06e0, 0x1, 0x1, 0xe446c0, ...)
	/home/jonas/.go/src/github.com/vmware/govmomi/property/collector.go:172 +0x694
github.com/vmware/govmomi/property.(*Collector).RetrieveOne(0xc4201cf908, 0x17dcb80, 0xc420014298, 0xc4202ab6a0, 0x18, 0xc4202ab740, 0x1b, 0xc4202c06e0, 0x1, 0x1, ...)
	/home/jonas/.go/src/github.com/vmware/govmomi/property/collector.go:200 +0x109
github.com/vmware/govmomi/object.Common.Properties(0xc4202c28a0, 0x15, 0xc42017f080, 0xc4202ab6a0, 0x18, 0xc4202ab740, 0x1b, 0x17dcb80, 0xc420014298, 0xc4202ab6a0, ...)
	/home/jonas/.go/src/github.com/vmware/govmomi/object/common.go:104 +0x134
github.com/vmware/govmomi/govc/dvs.(*add).Run(0xc4200171a0, 0x17dcb80, 0xc420014298, 0xc420062600, 0x0, 0x0)
	/home/jonas/.go/src/github.com/vmware/govmomi/govc/dvs/add.go:91 +0x3c5
github.com/vmware/govmomi/govc/cli.Run(0xc420010100, 0x4, 0x4, 0xc4200001a0)
	/home/jonas/.go/src/github.com/vmware/govmomi/govc/cli/command.go:156 +0x544
main.main()
	/home/jonas/.go/src/github.com/vmware/govmomi/govc/main.go:82 +0x64
```

As far as I understand hosts are created with 1 pnic, on which sits the default "vSwitch0", so there is no room to add the DVS, or am I missing some configuration?

--- a/govc/dvs/add.go
+++ b/govc/dvs/add.go
@@ -87,7 +87,7 @@ func (cmd *add) Run(ctx context.Context, f *flag.FlagSet) error {
 		return fmt.Errorf("%s (%T) is not of type %T", cmd.path, net, dvs)
 	}
 
-	var s mo.VmwareDistributedVirtualSwitch
+	var s mo.DistributedVirtualSwitch
 	err = dvs.Properties(ctx, dvs.Reference(), []string{"config"}, &s)
 	if err != nil {
 		return err

--- a/govc/test/network.bats
+++ b/govc/test/network.bats
@@ -156,6 +156,11 @@ load test_helper
   run govc dvs.create "$id"
   assert_success
 
+  local host=$GOVC_HOST
+
+  run govc dvs.add -dvs "$id" "$host"
+  assert_success
+
   run govc dvs.portgroup.add -dvs "$id" -type earlyBinding -nports 16 "${id}-ExternalNetwork"
   assert_success
 

--- a/simulator/folder.go
+++ b/simulator/folder.go
@@ -445,6 +445,33 @@ func (f *Folder) CreateDVSTask(req *types.CreateDVS_Task) soap.HasFault {
 			Description: spec.Description,
 		}
 
+		configInfo := &types.VMwareDVSConfigInfo{
+			DVSConfigInfo: types.DVSConfigInfo{
+				Uuid:                                dvs.Uuid,
+				Name:                                spec.Name,
+				ConfigVersion:                       spec.ConfigVersion,
+				NumStandalonePorts:                  spec.NumStandalonePorts,
+				MaxPorts:                            spec.MaxPorts,
+				UplinkPortPolicy:                    spec.UplinkPortPolicy,
+				UplinkPortgroup:                     spec.UplinkPortgroup,
+				DefaultPortConfig:                   spec.DefaultPortConfig,
+				ExtensionKey:                        spec.ExtensionKey,
+				Description:                         spec.Description,
+				Policy:                              spec.Policy,
+				VendorSpecificConfig:                spec.VendorSpecificConfig,
+				SwitchIpAddress:                     spec.SwitchIpAddress,
+				DefaultProxySwitchMaxNumPorts:       spec.DefaultProxySwitchMaxNumPorts,
+				InfrastructureTrafficResourceConfig: spec.InfrastructureTrafficResourceConfig,
+				NetworkResourceControlVersion:       spec.NetworkResourceControlVersion,
+			},
+		}
+
+		if spec.Contact != nil {
+			configInfo.Contact = *spec.Contact
+		}
+
+		dvs.Config = configInfo
+
 		if dvs.Summary.ProductInfo == nil {
 			product := Map.content().About
 			dvs.Summary.ProductInfo = &types.DistributedVirtualSwitchProductSpec{

